### PR TITLE
Allow custom content-type without setting a filename.

### PR DESCRIPTION
**Pull request by DylanPiercey**
Resolves: https://github.com/felixge/node-form-data/issues/74.

### CODE REVIEW

File: lib/form_data.js [hunk 1]
```diff
@@ -175,6 +175,8 @@ FormData.prototype._multiPartHeader = function(field, value, options) {
       options.contentType ||
       FormData.DEFAULT_CONTENT_TYPE
     );
+  } else if (options.contentType) {
```
Comment (alexindigo) [inline lib/form_data.js]: Can you add custom content-type for http streams too? Thank you.

File: test/integration/test-custom-content-type.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+// wrap non simple values into function
+// just to deal with ReadStream "autostart"
+// Can't wait for 0.10
+var FIELDS = {
+  'no_type': {
+    value: 'my_value'
+  },
+  'custom_type': {
+    value: 'my_value',
+    type: 'image/png',
```
Comment (alexindigo) [inline test/integration/test-custom-content-type.js]: Do you mind to rename it to `expectedType` so it'd be more clear on intent?

File: test/integration/test-custom-content-type.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+    req.on('data', function (data) { body += data.toString('utf-8'); });
+    req.on('end', function () {
+        // Separate body into individual files/fields and remove leading and trailing content.
+        var fields = body.split(boundry).slice(1, -1);
+
+        assert.ok(fields.length === 4);
+        
+        assert.ok(fields[0].indexOf('name="no_type"') > -1);
+        assert.ok(fields[0].indexOf('Content-Type"') === -1);
```
Comment (alexindigo) [inline test/integration/test-custom-content-type.js]: Nice solution :+1:

File: test/integration/test-custom-content-type.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    port: common.port,
+    path: '/'
+  }, function(err, res) {
+
+    if (err) {
+      throw err;
+    }
+
+    assert.strictEqual(res.statusCode, 200);
+
```
Comment (alexindigo) [inline test/integration/test-custom-content-type.js]: Would be nice to check that asserts above happened at all.

File: test/integration/test-custom-content-type.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        assert.ok(fields[1].indexOf('name="custom_type"') > -1);
+        assert.ok(fields[1].indexOf('Content-Type: ' + FIELDS.custom_type.type) > -1);
+        
+        assert.ok(fields[2].indexOf('name="default_type"') > -1);
+        assert.ok(fields[2].indexOf('Content-Type: ' + FIELDS.default_type.type) > -1);
+
+        assert.ok(fields[3].indexOf('name="implicit_type"') > -1);
+        assert.ok(fields[3].indexOf('Content-Type: ' + FIELDS.implicit_type.type) > -1);
+    });
+    res.end();
```
Comment (alexindigo) [inline test/integration/test-custom-content-type.js]: Better move it into `req.on('end')` so we'd answer to the client after we'd finish with whole request.

**Comment by alexindigo** [comment]
Thanks a lot for the PR, looks very nice. Just few minor changes and we'd hit home run. :)

File: lib/form_data.js (hunk 1 above)
Comment (DylanPiercey) [inline lib/form_data.js]: Each of the other parts of the "if else" chain contains a `|| FormData.DEFAULT_CONTENT_TYPE`. This should work for stream no?

Comment (DylanPiercey) [inline lib/form_data.js]: Fixed.

File: test/integration/test-custom-content-type.js (hunk 4 above)
Comment (DylanPiercey) [inline test/integration/test-custom-content-type.js]: Now tests that everything has ran.

**Comment by DylanPiercey** [comment]
Let me know if any other tweaks are needed.

**Pull request merged**